addons/project_purchase/models/project_project.py

```diff
@@ -28,22 +28,22 @@ def _compute_purchase_orders_count(self):
     # ----------------------------
 
     def action_open_project_purchase_orders(self):
-        purchase_orders = self.env['purchase.order.line'].search(
-            [('analytic_distribution', 'in', self.analytic_account_id.ids)],
+        purchase_order_lines = self.env['purchase.order.line'].search(
+            [('analytic_distribution', 'in', self.analytic_account_id.ids)]
         )
         action_window = {
             'name': _('Purchase Orders'),
             'type': 'ir.actions.act_window',
             'res_model': 'purchase.order',
             'views': [[False, 'tree'], [False, 'form']],
-            'domain': [('id', 'in', purchase_orders.ids)],
+            'domain': [('id', 'in', purchase_order_lines.order_id.ids)],
             'context': {
                 'project_id': self.id,
             }
         }
-        if len(purchase_orders) == 1:
+        if len(purchase_order_lines) == 1:
```

Comment:
maybe this should be:

```suggestion
        if len(purchase_order_lines.order) == 1:
```
in case there is several lines but of only one purchase orer

Reply:
Thank you for noticing this :ok_hand: